plugin/js/showpass-custom.js

```diff
@@ -285,14 +285,44 @@
 		const mutationObserver = new MutationObserver(function(mutations) {
 			mutations.forEach(function(mutation) {
 				if (mutation.target.className.includes('showpass-widget-body')) {
-					let gobj = window[window.GoogleAnalyticsObject];
-					let tracker, linker;
+
 					let iFrame = document.getElementById('showpass-widget');
+
+					// For analytics.js (UA)
+					// We use the linker provided by analytics.js to decorate the iframe src
+					let gobj = window[window.GoogleAnalyticsObject];
 					if (gobj) {
-						tracker = gobj.getAll()[0];
-						linker = new window.gaplugins.Linker(tracker);
+						let tracker = gobj.getAll()[0];
+						let linker = new window.gaplugins.Linker(tracker);
 						iFrame.src = linker.decorate(iFrame.src);
 					}
+
+					// For gtag.js (GA4)
+					// We use the gtag's get commands to get the client_id and session_id to decorate the iframe src
+					// This is additional to the analytics.js linker that should already be in place, AND the cross-domain tracking configured on the GA4 property itself
+					// which is for inbound/outbound clicks.
+					// @see https://support.google.com/analytics/answer/10071811?hl=en#zippy=%2Cmanual-setup
+					if (window.gtag && window.dataLayer) {
+						// Get the first available gtag config on the page. This config will be used
+						// to get the client_id and the session_id that we pass onto our iframe for our
+						// SDK to consume.
+						let ga4Config = window.dataLayer.find((x) => x[0] === "config" && x[1].startsWith("G-"));
+						let ga4Id = ga4Config[1];
+
+						let url = new URL(iFrame.src);
+
+						window.gtag('get', ga4Id, 'client_id', (client_id) => {
+							window.gtag('get', ga4Id, 'session_id', (session_id) => {
+								try {
+									url.searchParams.append('client_id', client_id);
+									url.searchParams.append('session_id', session_id);
+									iFrame.src = url.toString();
+								} catch(e) {
+									console.error(e)
+								}
```

Comment:
Will this all still work fro universal analytics? Or is it not backwards compatible?